models/models.articles.js

```diff
@@ -45,7 +44,14 @@ exports.selectArticles = (query) => {
 
 exports.selectArticleById = (id) => {
   return db
-    .query(`SELECT * FROM articles WHERE article_id=$1;`, [id])
+    .query(
+      `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.body, COUNT(comments.article_id)::int AS comment_count
+      FROM articles
+      LEFT JOIN comments ON articles.article_id=comments.article_id
+      WHERE articles.article_id=$1
+      GROUP BY articles.article_id;`,
+      [id]
+    )
     .then(({ rows, rowCount }) => {
       if (rowCount === 0) {
         return Promise.reject({
```

Comment:
following my previous comment, if below here in the model you return `rows[0]`  it will unnest the data from the array